### DISCUSSION TITLE
Why implement the engines in rust ?

### DISCUSSION THREAD

**Discussion by allyraza** [prisma/prisma#2348] [Q&A] [answered]
Hello, 

Very interesting project!

I came across prisma some time ago did not get a chance to use it until recently I started exploring other options to query builders (e.g. knex) and was going through the code and realize it uses binaries as child process to communicate to the database. 

I would very much like to know what are the benefits behind implementing the engines in rust other than rust is a compiled language and running them as child precess,  Would it not make it slow for an app to talk to a child process than that process talks to the database. 

-- Cheers

**Comment by ryands17** [comment, marked as answer]
Hey @allyraza :wave: 
The main reason for implementing the engines is Rust is that in Prisma 1 was implemented in Scala which means it needed the JVM to run. To reduce the overhead of running Prisma, we decided to rewrite it in Rust.

Benefits of Rust would be a significantly lower memory footprint and better performance than the JVM (or any other runtime like Node for that matter), and no more need to deploy, monitor and maintain an extra server to run Prisma 2.

We are also thinking of enhancing the bindings for Node with Rust [here](https://github.com/prisma/prisma/issues/767) which would significantly improve performance.

Hope that answers your question :)

**Reply by Sytten** [reply]
It should also be said that the GraphQL interface is now an implementation detail and not using the standard anyway, so maybe the narrative should be changed to `historical reasons` for the Rust choice? Also supporting multiple languages is also a good reason (there is a Go client in progress).

**Comment by ludralph** [comment]
Hi @allyraza,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
